Add DJANGO_CT, DJANGO_ID, ID to be used with '__exact' internally.

karolyi opened the pull request:
Hey,

somewhere around 8568e49b8f350ea3ce1358fad014c40b84f087b6 an exclusion has been added to `django_ct` with an `__exact` search. I need to add the other django internal fields excluded from text searches as well. 

In my use case I `.exclude(` some IDs for which I search for with the internal `id` field, that is a text field and did get escaped as a phrase because of that. My code uses `SearchQuerySet.exclude(id__list=['id_1', 'id_2']` which in the current state will not work.

These commits are about that, and also about preparing a new release. Please do a release ASAP, as I need these changes and can't use the official packages until this gets merged. I'm using my on gitea mirror for now, but would like to get back to using the official packages.

karolyi commented:
@claudep hey, can we pick up some pace here please?

claudep reviewed (changes requested):
Thanks for the patch. The release/version stuff will need to be in a separate PR dedicated to the next release. As for the rest of the changes, this will need a regression test.

karolyi commented:
I can remove the release, but I have no idea what you mean when you talk about the regression tests.

What I know though is, before the mentioned change in the commit, I could excloude `id`s, and now I can't.

karolyi commented:
I created a test for myself though (in my private system) that tests for this, maybe this can help:

```python
class HaystackXapianBackendTestCase(TestCase):
    'Testing haystack with xapian backend.'

    def test_exclude_by_id(self):
        'Test if the ID field is not phrased.'
        # See https://github.com/notanumber/xapian-haystack/pull/232
        sqs = SearchQuerySet().exclude(id__in=['testing123', 'testing456'])
        expected = \
            'Query((<alldocuments> AND_NOT (Qtesting123 OR Qtesting456)))'
        self.assertEqual(first=str(sqs.query.build_query()), second=expected)
```

claudep commented:
Yes, this is a good start. Now you will have to include that test somewhere in https://github.com/notanumber/xapian-haystack/tree/master/tests/xapian_tests/tests.

karolyi commented:
Alright, I can do that, if you can promise a timely release.

claudep commented:
Depends on what you call "timely", do not forget we are volunteers!

karolyi commented:
> do not forget we are volunteers!

yeah, so am I. good news: I've added the test, reverted the version bump, and squashed the commits into one single, clean commit. tests are running, we should be good to go!

claudep commented on .gitignore:
```diff
@@ -4,3 +4,4 @@
 dist
 NOTES
 MANIFEST
+venv
```
May I just ask you to remove this line? User-specific stuff should go in your own global excludes.

karolyi replied:
If I can object: this is not user specific but project specific, and should be harmless since venv is a general name for a locally used environment.

Not all user projects are python based.

claudep replied:
What I meant is the fact that storing a virtualenv inside a venv under the project source is your personal habit (which may be a good one, of course). But someone could decide to put in .venv, or in virtualenv or in .virtualenv or... you see what I mean. Read also https://git-scm.com/docs/gitignore:

`Patterns which a user wants Git to ignore in all situations (e.g., backup or temporary files generated by the user’s editor of choice) generally go into a file specified by core.excludesFile in the user’s ~/.gitconfig. Its default value is $XDG_CONFIG_HOME/git/ignore. If $XDG_CONFIG_HOME is either not set or empty, $HOME/.config/git/ignore is used instead.`

karolyi replied:
I feel like we're splitting hairs here, but so be it, I'll make an update soon.

karolyi replied:
Btw along your reasoning, removing the *.tmproj and _.DS_Store ignores should also apply.

claudep replied:
Absolutely!

karolyi replied:
Well, I can still modify the PR ;)

I really just want to get my changes released so I can use the offical package instead of my repo.

claudep reviewed (approved):
Thanks!

karolyi commented:
Nice, now all I need is a release.

Is that in your power to do?

claudep commented:
See PR #233
